Sources/ConsoleKit/Utilities/LoggerFragment.swift

```diff
@@ -0,0 +1,307 @@
+public struct LevelFragment: LoggerFragment {
+    public init() { }
+    
+    public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+        output += "[ \(record.level.name) ]".consoleText(record.level.style)
+        output.needsSeparator = true
+    }
+}
+
+/// Writes the given text to the output. This type does not request a separator for the next fragment
+public struct LiteralFragment: LoggerFragment {
+    public let literal: ConsoleText
+    
+    public init(_ literal: ConsoleText) {
+        self.literal = literal
+    }
+    
+    public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+        output += self.literal
+    }
+}
+
+public struct SeparatorFragment<T: LoggerFragment>: LoggerFragment {
+    public let literal: ConsoleText
+    public var fragment: T
+    
+    public init(_ literal: ConsoleText, fragment: T) {
+        self.literal = literal
+        self.fragment = fragment
+    }
+    
+    public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+        if output.needsSeparator {
+            if self.fragment.hasContent(record: &record) {
+                output.needsSeparator = false
+                output += self.literal
+            }
+        }
+        
+        self.fragment.write(&record, to: &output)
+    }
+}
+
+/// Writes the logged message to the output, and requests a separator for the next fragment.
+public struct MessageFragment: LoggerFragment {
+    public init() { }
+    
+    public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+        output += record.message.description.consoleText()
+        output.needsSeparator = true
+    }
+}
+
+/// Writes the combined metadata to the output, and requests a separator for the next fragment only if the metadata was not empty.
+///
+/// This fragment is considered to not have content if the metadata is empty.
+public struct MetadataFragment: LoggerFragment {
+    public init() { }
+    
+    public func hasContent(record: inout LogRecord) -> Bool {
+        !record.allMetadata().isEmpty
+    }
+    
+    public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+        let allMetadata = record.allMetadata()
+        
+        guard !allMetadata.isEmpty else { return }
+        
+        output += allMetadata.sortedDescriptionWithoutQuotes.consoleText()
+        output.needsSeparator = true
+    }
+}
+
+/// Writes the file location of the logged message, including the line. This fragment requests a separator for the next fragment.
+public struct SourceLocationFragment: LoggerFragment {
+    public init() { }
+    
+    public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+        let file = record.file + ":" + record.line.description
+        output += "(" + file.consoleText() + ")"
+        output.needsSeparator = true
+    }
+}
+
```

Comment:
Can we also add a `SourceModuleFragment` for rendering the `source` field of the `LogRecord`? It shouldn't be part of the default, but we should provide it.